proposals/bip-002.md

```diff
@@ -0,0 +1,93 @@
+# Explicit Annotated status and completeness information
+
+| Field | Value |
+| --- | --- |
+| BIP | 2 |
+| Title | Explicit Annotated status and completeness information |
+| Author | Juanjo Álvarez |
+| Status | Draft |
+| Created | 2017-06-01 |
+| Updated | 2017-06-01 |
+
+## Abstract
+
+This BIP proposed a new official (as in defined in the Node struct and
+documented) "AnnotationStatus" property that would mark if the role has been
+annotated in any way and the quality of the current annotation.
```

Comment:
The complete / incomplete / un annotated status mark nodes, not roles (even tough the nodes could have roles). 

Example: a language has a normal function definition that could be mapped to FunctionDef without problems  but also some variant that is somewhat different (for example an async function def), since the async / await nodes are not in the UAST the same driver would annotate as "incomplete" only the second one.